galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/AccessCps.java

```diff
@@ -54,10 +56,54 @@
     public IFramework framework; // NOSONAR
 
     @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        String[] reqParams = req.getPathInfo().substring(1).split("/");
-        if(reqParams.length == 1 && reqParams[0].equals("namespace")) {
-            JsonArray namespaceArray = new JsonArray();
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException {
+        try {
+            Pattern pattern1 = Pattern.compile("/namespace/?");
+            Matcher matcher1 = pattern1.matcher(req.getPathInfo());
+            Pattern pattern2 = Pattern.compile("/namespace/([A-z0-9]+)/?");
+            Matcher matcher2 = pattern2.matcher(req.getPathInfo());
+            Pattern pattern3 = Pattern.compile("/namespace/([A-z0-9]+)/prefix/([A-z0-9._\\-]+)/suffix/([A-z0-9._\\-]+)/?");
+            Matcher matcher3 = pattern3.matcher(req.getPathInfo());
```

Comment:
these matchers are expensive operations,  so you don't want to run all of them every get.   So put the one that is likely to be most common first:-
matcher = 
if (matched) {
call
return
}
matcher
if ( .....